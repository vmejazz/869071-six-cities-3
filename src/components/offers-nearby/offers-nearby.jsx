import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import ApartmentList from "../apartment-list/apartment-list.jsx";
import {Operation} from "../../reducer/data/data.js";
import {getOffersNearby} from "../selectors.js";

class OffersNearby extends PureComponent {

  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {loadNearby, activeOfferId} = this.props;

    loadNearby(activeOfferId);
  }

  // componentWillReceiveProps() {
  //   const {loadNearby, activeOfferId} = this.props;

  //   loadNearby(activeOfferId);

  // }

  componentWillReceiveProps(nextProps) {
    const {loadNearby, activeOfferId: nextId} = nextProps;
    const {activeOfferId: oldID} = this.props;

    if (oldID !== nextId) {
      loadNearby(nextId);
    }
  }


  render() {
    const {offersNearby = []} = this.props;

    return (
      <ApartmentList
        offersShow={offersNearby}
      />
    );
  }
}

OffersNearby.propTypes = {
  loadNearby: PropTypes.func,
  activeOfferId: PropTypes.string.isRequired,
  offersNearby: PropTypes.arrayOf(PropTypes.object),
};

const mapStateToProps = (state) => ({
  offersNearby: getOffersNearby(state),
});

const mapDispatchToProps = (dispatch) => ({
  loadNearby(offerId) {
    dispatch(
        Operation.loadNearby(offerId)
    );
  },
});

export {OffersNearby};
export default connect(mapStateToProps, mapDispatchToProps)(OffersNearby);
