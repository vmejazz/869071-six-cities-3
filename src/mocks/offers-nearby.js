const offersNearby = [
  {
    id: 1,
    title: `Beautiful apartment`,
    price: 120,
    srcImg: `img/apartment-01.jpg`,
    srcGallery: [`img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `All super long text about apartmmnet with cools stuffs`,
    premium: false,
    type: `Room`,
    rate: 4.8,
    bedrooms: 3,
    maxGuests: 4,
    apartmentStuff: [`Wifi`, `Cable TV`, `Kitchen`],
    ownerInfo: {
      name: `Jon`,
      super: true,
      srcAvatar: `img/avatar-max.jpg`
    },
    position: [52.3909553943508, 4.85309666406198],
    city: `Amsterdam`,
    reviews: [
      {
        srcAvatar: `img/avatar-max.jpg`,
        name: `Max`,
        rate: 3.4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        date: `2019-04-24`
      },
      {
        srcAvatar: `img/avatar-max.jpg`,
        name: `Nikola`,
        rate: 4.6,
        text: `A very comfertable apartments for rest and relax.`,
        date: `2018-04-24`
      }
    ]
  },
  {
    id: 2,
    title: `Perfect apartment`,
    price: 200,
    srcImg: `img/apartment-02.jpg`,
    srcGallery: [`img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `All super long text about apartmmnet with cools stuffs`,
    premium: false,
    type: `House`,
    rate: 1,
    bedrooms: 3,
    maxGuests: 4,
    apartmentStuff: [`wifi`, `Cable TV`, `Kitchen`],
    ownerInfo: {
      name: `Jon`,
      super: true,
      srcAvatar: `img/avatar-max.jpg`
    },
    position: [50.938014, 6.958104],
    city: `Cologne`,
    reviews: [
      {
        srcAvatar: `img/avatar-max.jpg`,
        name: `Max`,
        rate: 3.4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        date: `2019-04-24`
      },
      {
        srcAvatar: `img/avatar-max.jpg`,
        name: `Nikola`,
        rate: 4.6,
        text: `A very comfertable apartments for rest and relax.`,
        date: `2018-04-24`
      }
    ]
  },
  {
    id: 3,
    title: `Imagine apartment`,
    price: 250,
    srcImg: `img/apartment-03.jpg`,
    srcGallery: [`img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `All super long text about apartmmnet with cools stuffs`,
    premium: false,
    type: `Apartment`,
    rate: 3,
    bedrooms: 3,
    maxGuests: 4,
    apartmentStuff: [`wifi`, `Cable TV`, `Kitchen`],
    ownerInfo: {
      name: `Jon`,
      super: true,
      srcAvatar: `img/avatar-max.jpg`
    },
    position: [52.3909553943508, 4.929309666406198],
    city: `Amsterdam`,
    reviews: [
      {
        srcAvatar: `img/avatar-max.jpg`,
        name: `Max`,
        rate: 3.4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        date: `2019-04-24`
      },
      {
        srcAvatar: `img/avatar-max.jpg`,
        name: `Nikola`,
        rate: 4.6,
        text: `A very comfertable apartments for rest and relax.`,
        date: `2018-04-24`
      }
    ]
  }
];

export default offersNearby;
