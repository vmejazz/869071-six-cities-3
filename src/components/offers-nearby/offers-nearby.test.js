import React from "react";
import renderer from "react-test-renderer";
import {OffersNearby} from "./offers-nearby.jsx";
jest.mock(`../apartment-list/apartment-list.jsx`, () => `ApartmentList`);

it(`<OffersNearby /> should render `, () => {
  const tree = renderer
      .create(
          <OffersNearby
            loadNearby={() => {}}
            activeOfferId={`5`}
            offersNearby={[]}
          />
      )
      .toJSON();

  expect(tree).toMatchSnapshot();
});
