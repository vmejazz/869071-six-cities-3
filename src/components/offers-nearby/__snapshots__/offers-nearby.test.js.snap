// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OffersNearby /> should render  1`] = `
<ApartmentList
  offersShow={Array []}
/>
`;
